src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -143,11 +143,11 @@ bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
         if (ourNode->isLeaf () && otherNode->isLeaf ())
         {
             // two leaves
-            if (ourNode->getTag () == otherNode->getTag ())
+            if (ourNode->peekItem()->getTag () == otherNode->peekItem()->getTag ())
```

Comment:
The term "peek" in rippled means "look at private data members". I'm not saying this is a good thing (it isn't). I agree that this is a bit of ugliness. SHAMap could use a spring cleaning in terms of its interfaces. But if it helps you sleep at night, at least with this change there are fewer public members.